### ISSUE TITLE
delete --all crashes 

### ISSUE THREAD

**Issue by vxbinaca** [jjjake/internetarchive#229]
`ia delete satoshi-p2p-research-mirror --all `

```
Deleting files from satoshi-p2p-research-mirror
Traceback (most recent call last):
  File "/usr/local/bin/ia", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.5/dist-packages/internetarchive/cli/ia.py", line 159, in main
    sys.exit(ia_module.main(argv, session))
  File "/usr/local/lib/python3.5/dist-packages/internetarchive/cli/ia_delete.py", line 137, in main
    headers=args['--headers'],
KeyError: '--headers'
```

Please close if duplicate for some reason I remember running into this before and possibly making an issue out of it.

Also, how do I set delete on upload in the CLI? Your documentation isn't clear how to do that. It's a library call but not a command line thing. Or maybe I'm making a mistake.

**Comment by jjjake** [comment]
Thanks for the report @vxbinaca! I'll take a look.

To delete files after they've been successfully uploaded (i.e. a 200 is received and a checksum calculated by S3 is compared with one calculated locally):

```ia upload --delete <id> <file>... ...```

Does that make sense?

**Comment by vxbinaca** [comment]
Makes sense, just the bug is all I'm really after to be fixed. Thanks.

I have exciting things I'm saving right now. Working with Jason on something neat. Found some hot archival wares with even more important things planned for the future.

**Comment by jjjake** [comment]
Turns out this bug was causing all ``ia delete`` requests to fail due to the typo, so double thanks for catching that and reporting!

I just patched it: https://github.com/jjjake/internetarchive/commit/f70b09858ce2c1b7c7ac97037e41d2fe6e633fba

I hope to ship it tomorrow (there's another feature that I'm hoping to ship with it, but it needs a little more testing). It's also a single character change, so it should be able to patch your local copy in the meantime

**Comment by vxbinaca** [comment]
Polish the other feature because this isn't vital to me right now. Although I've experienced this bug for more than a year, but I feel like I made a report before or mentioned something somewhere.

Should have piped up earlier. Anyway, glad I could help. Closing.

**Comment by jjjake** [comment]
Hmm, that must have been a different bug. This bug was introduced 29 days ago: https://github.com/jjjake/internetarchive/blob/347306ad83eca7c40a82617b221389c350ead69e/internetarchive/cli/ia_delete.py#L136

The other feature is polished, I just want to do some final testing. I'll probably ship v1.7.6 tomorrow that will have this bug fix in their. Thanks again.

**Comment by jjjake** [comment]
I just shipped v1.7.6, @vxbinaca.
